tests/cases/arithmetic/shift_right.test

```diff
@@ -0,0 +1,14 @@
+### SUBSTRAIT_SCALAR_TEST: v1.0
+### SUBSTRAIT_INCLUDE: '/extensions/functions_arithmetic.yaml'
+
+# basic: Basic examples without any special cases
```

Comment:
nit I was wondering if we should cover shifts with 0 -- but this could apparently be a special case.